src/session_impl.cpp

```diff
@@ -1467,7 +1461,7 @@ namespace aux {
 					!= m_settings.get_int(settings_pack::ssl_listen))
 			||
 #endif
-			(pack.has_val(settings_pack::listen_interfaces)
+			init || (pack.has_val(settings_pack::listen_interfaces)
```

Comment:
instead of injecting "init" here, you would get the same semantics by putting it in the if (reopen_listen_port) condition, right? (that seems like it would make things a bit simpler, this is already a complex expression)